lib/emscripten/src/lib.rs

```diff
@@ -351,10 +351,32 @@ pub fn run_emscripten_instance(
     // println!("running emscripten instance");
 
     if let Some(ep) = entrypoint {
-        debug!("Running entry point: {}", &ep);
-        let arg = unsafe { allocate_cstr_on_stack(instance.context_mut(), args[0]).0 };
-        //let (argc, argv) = store_module_arguments(instance.context_mut(), args);
-        instance.call(&ep, &[Value::I32(arg as i32)])?;
+        let func_name = &ep;
+        debug!("Running entry point: {}", func_name);
+        let func = instance.dyn_func(func_name)?;
+
+        let num_params = func.signature().params();
+        debug!("Expected parameters: {:?}", num_params);
+
+        if num_params.len() != args.len() {
+            return Err(CallError::Resolve(ResolveError::ExportWrongType {
+                name: func_name.to_string(),
+            }));
+        }
+
+        let mut new_args: Vec<Value> = Vec::new();
+        for arg in args.iter() {
+            match arg.parse() {
+                Err(_e) => {
+                    return Err(CallError::Resolve(ResolveError::ExportWrongType {
+                        name: func_name.to_string(),
+                    }))
+                }
+                Ok(x) => new_args.push(Value::I32(x)),
+            }
+        }
```

Comment:
⛳️ Consider doing this functionally. I personally find this style easier to read and it doesn't leak the mutability of `new_args`, but this is a very minor suggestion and "no, the for loop is easier to read" is a fine answer.

```Rust
let new_args = args
                   .iter()
                   .map(|a| a.parse().map_err(|_| CallError::Resolve(ResolveError::ExportWrongType { name: func_name.to_string() }).map(Value::I32))
                   .collect::<Result<Vec<_>,_>>()?;
```

Reply:
No prob ;)